.github/workflows/full-qa.yml

```diff
@@ -0,0 +1,135 @@
+name: Full QA
+
+on:
+  # new commits pushed to release branches
+  push:
+    branches:
+      - "master" # pushes to the `master` branch
+      - "*.*" # pushes to the released version branches (e.g. `1.0`)
+
+jobs:
+  unit-tests-p2:
+    name: Unit Tests (Polymer 2)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Node 12.x
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12.x
+
+      - name: Check out the source code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Install global npm dependencies
+        # bower is needed to run 'bower install'
+        # polymer-cli is needed to run the lint step
+        # web-component-tester is needed to run the test step
+        run: "npm install --quiet --no-progress --global bower polymer-cli web-component-tester"
+
+      - name: Install project npm dependencies
+        run: "npm install --quiet --no-progress"
+
+      - name: Install project Bower dependencies
+        run: "bower install --quiet"
+
+      - name: Run automated magi-cli checks
+        run: "npm run check"
+
+      - name: Run a linter
+        run: "npm run lint"
+
+      - name: Run unit tests on SauceLabs
+        if: ${{ github.event != 'schedule' }}
+        run: "wct --env saucelabs"
+        env:
+          SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
+          SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+
+      - name: Run unit tests on SauceLabs (cron)
+        if: ${{ github.event == 'schedule' }}
+        run: "wct --env saucelabs-cron"
+        env:
+          SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
+          SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+
+  unit-tests-p3:
+    name: Unit Tests (Polymer 3)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Node 12.x
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12.x
+
+      - name: Check out the (Polymer 2) source code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Install global npm dependencies
+        # magi-cli, bower and polymer-modulizer are needed to run the Polymer 3 conversion step
+        # web-component-tester is needed to run the test step
+        run: "npm install --quiet --no-progress --global bower magi-cli web-component-tester polymer-modulizer"
+
+      - name: Convert the source code to Polymer 3
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          magi p3-convert --out . --import-style=name
+
+      - name: Install project npm dependencies
+        run: "yarn install --flat"
```

Comment:
We probably should add a comment above explaining that yarn is needed to check for correct versions.
This is how we check that two versions of a component are never installed (it would fail the build).

Reply:
done